core/src/datayoga_core/blocks/http/write/block.py

```diff
@@ -0,0 +1,125 @@
+        if connection_details["type"] != "http":
+            raise ValueError(f"{connection_name} is not an HTTP connection")
+
+        self.base_uri = f"{connection_details['base_uri'].rstrip('/')}"
+        self.method = self.properties["method"]
+        output = self.properties.get("output", {})
+        self.response_status_code_field = output.get("status_code")
+        self.response_headers_field = output.get("headers")
+        self.response_content_field = output.get("body")
+        self.timeout = self.properties.get("timeout", connection_details.get("timeout", 10))
+
+        def process_dict(input_dict, output_dict):
+            for key, value in input_dict.items():
+                if isinstance(value, dict) and "expression" in value and "language" in value:
+                    output_dict[key] = expression.compile(value["language"], value["expression"])
+                elif isinstance(value, dict):
+                    output_dict[key] = {}
+                    process_dict(value, output_dict[key])
+                else:
+                    output_dict[key] = value
+
+        self.request_config = {}
+        process_dict({
+            "endpoint": self.properties["endpoint"],
+            "headers": {**connection_details.get("headers", {}), **self.properties.get("extra_headers", {})},
+            "query_params": {
+                **connection_details.get("query_parameters", {}),
+                **self.properties.get("extra_query_parameters", {})},
+            "payload": self.properties.get("payload", {})
+        }, self.request_config)
+
+    async def run(self, data: List[Dict[str, Any]]) -> BlockResult:
+        logger.debug(f"Running {self.get_block_name()}")
+
+        block_result = BlockResult()
+
+        def process_dict(input_dict, output_dict):
+            for key, value in input_dict.items():
+                if isinstance(value, Expression):
+                    output_dict[key] = value.search_bulk(data)
+                elif isinstance(value, dict):
+                    output_dict[key] = {}
+                    process_dict(value, output_dict[key])
+                else:
+                    output_dict[key] = [value] * len(data)
+
+        request_configs = {}
+        process_dict(self.request_config, request_configs)
+
+        for i, row in enumerate(data):
+            response = None
+
+            try:
+                url = f"{self.base_uri}/{request_configs['endpoint'][i].lstrip('/')}"
+                headers = {key: value[i] for key, value in request_configs["headers"].items()}
+                query_params = {key: value[i] for key, value in request_configs["query_params"].items()}
+                payload = {key: value[i] for key, value in request_configs["payload"].items()}
+
+                logger.debug(
+                    f"Sending HTTP {self.method} request to: {url}\nheaders:{headers}\n\tquery_params: {query_params}\n\tpayload: {payload}")
+
+                if self.method == "GET":
```

Comment:
And… why not `aiohttp` or `httpx` which are async ?

Reply:
Done.